### DISCUSSION TITLE
[V3] Cannot read properties of null (reading 'before') Error when using Blade Component inside loop

### DISCUSSION THREAD

**Discussion by achmedislamic** [livewire/livewire#6663] [7. Bugs (archived)]
### Livewire version

3.0.1

### Which PHP version are you using?

PHP 8.1

### Repository URL

https://github.com/achmedislamic/livewire-error

### Steps To Reproduce

1. Create Livewire Component. We called it ÜserTable
2. Create 10 or more dummy data in user table in the database
3. Create anonymous blade component called tr.blade.php. We're using this instead of plain HTML tr tag to trigger the error.
4. Copy & Paste this code into your UserTable class

```
<?php

namespace App\Livewire;

use App\Models\User;
use Livewire\Component;

class UserTable extends Component
{
    public $search = '';

    public function render()
    {
        return view('livewire.user-table', [
            'users' => User::query()
                ->where('name', 'like', "%{$this->search}%")
                ->get()
        ]);
    }
}
```
5. Copy & Paste this code into your user-table.blade.php

```
<div>
    <input type="search" wire:model.live="search">
    <table>
        <tr>
            <th>Nama</th>
        </tr>
        @foreach ($users as $user)
            <x-tr>{{-- error when using blade component --}}
                <td>{{ $user->name }}</td>
            </x-tr>

            {{-- not error when using plain HTML tag like this --}}
            {{-- <tr>
                <td>{{ $user->name }}</td>
            </tr> --}}
        @endforeach
    </table>
</div>
```

6. Visit the UserTable component.
7. Start typing in search input until the total of record change because of your typing.
8. Then delete all of your search query.
9. Error will be triggered in the browser console.

https://github.com/livewire/livewire/assets/25688207/16170360-d5f1-45fb-a5ea-e4992cb29840

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by achmedislamic** [comment]
apparently, this error occurred because in the tr.blade.php Blade Component, I wrote an inappropriate end tag like this:

```
<tr>
{{  $slot }}
</div>
```

I don't understand whether this error that I wrote are a bug or not. But, I let this discussion remain. Again, this is my mistake.

**Reply by gitadminhub** [reply]
Thank you, same problem and solution. It seems we are focusing too much on the backend after all. **said after couple of hours reading php lines**

Think that livewire could not track HTML DOM Elements because of the unclosed tags.

**Comment by chillbram** [comment]
Had a similar issue that was caused by not having a `<tr></tr>` tag around a `<td colspan="3"></td>` when defining the empty state for a `@forelse` statement. Not entirely sure why this would cause Livewire to break, but it resolved itself after fixing the HTML table logic.

**Comment by iRaziul** [comment]
This is why we should add `wire:key` to the opening tag of the first element within the loop. [Docs](https://livewire.laravel.com/docs/troubleshooting#adding-wirekey)
